db/migrate/20240206150348_create_comparison_metric_models.rb

```diff
@@ -0,0 +1,80 @@
+class CreateComparisonMetricModels < ActiveRecord::Migration[6.1]
+  def change
+    create_table :comparison_periods do |t|
+      t.string :label, null: false # we might want to translate this?
+      t.date :start_date, null: false
+      t.date :end_date, null: false
+
+      t.timestamps
+    end
+
+    create_table :comparison_metrics do |t|
+      t.references :school, null: false, foreign_key: { on_delete: :cascade }
+      t.references :metric_type, null: false, foreign_key: { to_table: 'comparison_metrics', on_delete: :cascade }
+      t.references :alert_type, null: false
+
+      t.string :value # type can vary, how do we do this?????
+
+      t.integer :reporting_period # enum
+      # thought splitting the periods in to two might provide more flexibility. I might be wrong but is
+      # it true that we're not always comparing to a previous period? and that a metric can just be relevant to
+      # just a single period?
+      # is there the option to store exact date information for every metric rather than just for custom ones?
```

Comment:
this is largely covered by the additional columns in the metric, as you can derive the dates from the asof_date for the fixed period.